File: packages/typescript/src/baseline.ts
import type { Grid } from "./types.ts";

/**
 * The "vanilla" un-compressed representation of a grid: rows joined with ` | `,
 * separated by `\n`, no escaping, no address prefixes. This is the form
 * `rawBaseline.tokenEstimate` is measured against — i.e. the raw text a
 * developer would otherwise paste into a prompt.
 */
export function vanillaEncode(grid: Grid): string {
  return grid.rows.map((row) => row.join(" | ")).join("\n");
}

File: packages/python/sheet_compressor/baseline.py
"""Vanilla raw-baseline encoding. See SPEC §7."""

from typing import Any, Mapping


def vanilla_encode(grid: Mapping[str, Any]) -> str:
    """Un-compressed baseline: rows joined with ' | ', separated by '\\n'."""
    rows = grid.get("rows", [])
    return "\n".join(" | ".join(row) for row in rows)
